### PR TITLE
Add Data Center Visualization

### PR DESCRIPTION
**Pull request by szok**
Connected to #1426

### CODE REVIEW

**Comment by mkurek** [comment]
what about https://github.com/allegro/ralph/issues/1451 ? Are we going to merge it as it is (just a copy from old ralph) and then adjust everything to support gulp etc or could we handle both topics in one PR? Or maybe should we first introduce gulp etc into NG and then move visualisation to NG? @ar4s @szok @xliiv ?

**Comment by vi4m** [comment]
Rule of thumb: Keep issue/stories _thin and lean_, but make architecture great.

Integrating gulp as we've seen is not the smallest task, so maybe start with something actually working and move fast in next interations?

File: src/ralph/assets/tests/factories.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    AssetModel,
+    Environment,
+    Service,
+    ServiceEnvironment
+)
+
+
+class AssetModelFactory(DjangoModelFactory):
+
+    name = factory.Sequence(lambda n: 'Environment #{}'.format(n))
```
Comment (mkurek) [inline src/ralph/assets/tests/factories.py]: Environment?

File: src/ralph/assets/tests/factories.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
```
Comment (mkurek) [inline src/ralph/assets/tests/factories.py:1]: BTW to discuss if we want to reuse these factories in demo data generation process. If yes, then we should probably give them meaningful names etc. @ar4s @xliiv ?

File: src/ralph/data_center/models.py [hunk 3]
```diff
@@ -160,6 +160,7 @@ class Meta:
         verbose_name_plural = _('VIPs')
 
 
+# TODO!!!
```
Comment (mkurek) [inline src/ralph/data_center/models.py]: TODO what?

File: src/ralph/data_center/tests/factories.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    DataCenterAsset,
+    Rack,
+    RackAccessory,
+    ServerRoom,
+)
+
+
+class DataCenterFactory(DjangoModelFactory):
+
+    name = factory.Sequence(lambda n: 'Data Center #{}'.format(n))
```
Comment (ar4s) [inline src/ralph/data_center/tests/factories.py]: use cycle to generate name https://github.com/allegro/ralph_assets/blob/94effeee0195b76ed25f1b1abfeed83765ebfb18/src/ralph_assets/tests/utils/assets.py 
https://github.com/allegro/ralph_assets/blob/94effeee0195b76ed25f1b1abfeed83765ebfb18/src/ralph_assets/tests/utils/assets.py#L62

File: src/ralph/dc_view/urls/api.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+from django.conf.urls import patterns, url
+
+from ralph.dc_view.views.api import (
+    DCAssetsView,
+    DCRacksAPIView
+)
+
+
+urlpatterns = patterns(
```
Comment (mkurek) [inline src/ralph/dc_view/urls/api.py]: See
https://docs.djangoproject.com/en/1.8/topics/http/urls/#syntax-of-the-urlpatterns-variable and https://docs.djangoproject.com/en/1.8/releases/1.8/#django-conf-urls-patterns - from Django 1.8 `patterns` is deprecated.

File: src/ralph/dc_view/urls/ui.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from django.conf.urls import patterns, url
+
+from ralph.dc_view.views.ui import DataCenterView
+
+
+urlpatterns = patterns(
```
Comment (mkurek) [inline src/ralph/dc_view/urls/ui.py]: ditto

File: src/ralph/assets/tests/factories.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    name = factory.Iterator(['3310', 'XD300S', 'Hero 3', 'Computer set'])
+
+    class Meta:
+        model = AssetModel
+
+
+class DataCenterAssetModelFactory(DjangoModelFactory):
+
+    name = factory.Iterator(['DL360', 'DL380p', 'DL380', 'ML10', 'ML10 v21'])
```
Comment (mkurek) [inline src/ralph/assets/tests/factories.py:30]: what will happen if I'll create (in the same test) 6 data center assets (which should reference `DataCenterAssetModelFactory` using `SubFactory`)?

File: src/ralph/assets/tests/factories.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        model = AssetModel
+        django_get_or_create = ['name']
+
+
+class EnvironmentFactory(DjangoModelFactory):
+
+    name = factory.Iterator(['prod', 'dev', 'test'])
+
+    class Meta:
+        model = Environment
```
Comment (mkurek) [inline src/ralph/assets/tests/factories.py:42]: imo should have `django_get_or_create` as well

Comment (szok) [inline src/ralph/assets/tests/factories.py:42]: Done

**Comment by mkurek** [comment]
:ok:

File: src/ralph/data_center/models/physical.py [hunk 9]
```diff
@@ -129,17 +219,38 @@ class DataCenterAsset(Asset):
...
-        symmetrical=False,
-    )
+    def get_related_assets(self):
+        """Returns the children of a blade chassis"""
+        orientations = [Orientation.front, Orientation.back]
+        assets_by_orientation = []
+        for orientation in orientations:
+            assets_by_orientation.append(list(
+                DataCenterAsset.objects.select_related('model').filter(
+                    position=self.position,
```
Comment (ar4s) [inline src/ralph/data_center/models/physical.py]: try replace `position=self.position, rack=self.rack` by `parent=self`

**Comment by ar4s** [comment]
:ok:

**Pull request merged**